internal/controllers/cryostat_controller_test.go

```diff
@@ -2593,20 +2615,21 @@ func (t *cryostatTestInput) checkMainPodTemplate(deployment *appsv1.Deployment,
 	emptyDir := cr.Spec.StorageOptions != nil && cr.Spec.StorageOptions.EmptyDir != nil && cr.Spec.StorageOptions.EmptyDir.Enabled
 	builtInDiscoveryDisabled := cr.Spec.TargetDiscoveryOptions != nil && cr.Spec.TargetDiscoveryOptions.BuiltInDiscoveryDisabled
 	dbSecretProvided := cr.Spec.JmxCredentialsDatabaseOptions != nil && cr.Spec.JmxCredentialsDatabaseOptions.DatabaseSecretName != nil
+
 	checkCoreContainer(&coreContainer,
 		t.minimal, t.TLS, t.externalTLS, t.EnvCoreImageTag,
 		t.controller.IsOpenShift, ingress, reportsUrl,
 		cr.Spec.AuthProperties != nil, emptyDir, builtInDiscoveryDisabled, dbSecretProvided,
-		cr.Spec.Resources.CoreResources, test.NewCoreSecurityContext(cr))
+		*test.NewCoreContainerResource(cr), test.NewCoreSecurityContext(cr))
```

Comment:
I would change the resources parameter for these functions to be a pointer instead of dereferencing when calling them.